src/odemis/acq/acqmng.py

```diff
@@ -83,6 +85,205 @@ def acquire(streams, settings_obs=None):
     return future
 
 
+def acquireZStack(streams, zlevels, settings_obs=None):
+    """
+    The acquisition manager of a zstack
+    streams (list of Stream): the streams to be acquired
+    zlevels (dict Stream -> list of floats): a dictionary containing the streams and the
+    corresponding lists of actuator z positions
+    settings_obs (SettingsObserver): VAs to be integrated in the acquired data as metadata
+    return (ProgressiveFuture): the future that will be executing the task
+    """
+    # create future
+    future = model.ProgressiveFuture()
+    # create acquisition task
+    acqui_task = ZStackAcquisitionTask(future, streams, zlevels, settings_obs)
+    # add the ability of cancelling the future during execution
+    future.task_canceller = acqui_task.cancel
+
+    # set the progress of the future
+    total_duration = acqui_task.estimate_total_duration()
+    future.set_end_time(time.time() + total_duration)
+
+    # assign the acquisition task to the future
+    executeAsyncTask(future, acqui_task.run)
+
+    return future
+
+
+class ZStackAcquisitionTask(object):
+    """
+    This class represents an acquisition task for a zstack of data.
+    """
+
+    def __init__(self, future, streams, zlevels, settings_obs):
+        """
+        The class constructor
+        future (ProgressiveFuture): the future that will execute the task asynchronously
+        streams (list of Stream): the streams to be acquired
+        zlevels (dict Stream -> list of floats): a dictionary containing the streams and the
+        corresponding lists of actuator z positions
+        settings_obs (SettingsObserver): VAs to be integrated in the acquired data as metadata
+        """
+        self._main_future = future
+        self._future_lock = threading.Lock()
+        self._actuator_f = None 
+        self._single_acqui_f = None 
+        self._future_state = None
+        self._streams = sortStreams(streams)
+        self._zlevels = zlevels
+        self._settings_obs = settings_obs
+        self._zpos = list(zlevels.values())[0] if zlevels.values() else None
+        zstep = abs(self._zpos[0]-self._zpos[1]) if self._zpos else 0
+        # get the focuser object of any of the FM streams
+        focuser = next((s.focuser for s in self._streams if isinstance(s, FluoStream)), None)
+        self._zstep_duration = estimate_focuser_move_duration(focuser, zstep)
+
+    def cancel(self, _):
+        """
+        Called when the user presses the button "cancel" and the future is still running.
```

Comment:
It's better to not write the docstring about how currently it's used, but what it does and when it should be called (in general). So here I'd say "Cancels the acquisition task". You can mention it's the canceller of the ProgressiveFuture, as it's fixed by the class, but let's not mention the GUI.